core/core-impl/src/main/java/ai/timefold/solver/core/api/solver/SolverManager.java

```diff
@@ -98,6 +99,65 @@ static <Solution_, ProblemId_> SolverManager<Solution_, ProblemId_> create(
         return new DefaultSolverManager<>(solverFactory, solverManagerConfig);
     }
 
+    // ************************************************************************
+    // Builder methods
+    // ************************************************************************
+
+    /**
+     * Creates a Builder that allows to customize and submit a planning problem to solve.
+     *
+     * @param problemId never null, a ID for each planning problem. This must be unique.
+     *        Use this problemId to {@link #terminateEarly(Object) terminate} the solver early,
+     * @param problem never null, a {@link PlanningSolution} usually with uninitialized planning variables
+     * @return never null
+     */
+    default SolverJobFactory<Solution_, ProblemId_> builder(ProblemId_ problemId, Solution_ problem) {
+        return builder(problemId, (problemId_) -> problem);
+    }
+
+
+    /**
+     * As defined by {@link #builder(Object, Object)}
+     *
+     * @param problemId never null, a ID for each planning problem. This must be unique.
+     *        Use this problemId to {@link #terminateEarly(Object) terminate} the solver early,
+     *        {@link #getSolverStatus(Object) to get the status} or if the problem changes while solving.
+     * @param problemFinder never null, function that returns a {@link PlanningSolution}, usually with uninitialized planning
+     *                      variables
+     * @return never null
+     */
+    SolverJobFactory<Solution_, ProblemId_> builder(ProblemId_ problemId,
+                                                    Function<? super ProblemId_, ? extends Solution_> problemFinder);
```

Comment:
You have no way of knowing this, so just for context: we hate the fact that the user has to provide `problemId`. It complicates the API, forces the user to have to think about things they may not need.

How about creating another builder overload which doesn't require a `problemId` and instead generates `long` IDs internally? We can not get rid of the `problemId`, but we can at least help hide it if the user doesn't care.